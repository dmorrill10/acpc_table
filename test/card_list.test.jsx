import React from 'react';
import renderer from 'react-test-renderer';
const {CardList, BoardCards, PrivateCards} = require('../dist/card_list.js');

test('with no cards works', () => {
  const cards = [];
  const patient = <CardList cards={cards}/>;
  const tree = renderer.create(patient).toJSON();
  expect(tree).toMatchSnapshot();
});
test('with two face-down cards works', () => {
  const cards = [{}, {}];
  const patient = <CardList cards={cards}/>;
  const tree = renderer.create(patient).toJSON();
  expect(tree).toMatchSnapshot();
});
test('with two face-up cards works', () => {
  const cards = [
    {
      rank: '2',
      suit: '&hearts;'
    }, {
      rank: '3',
      suit: '&diamonds;'
    }
  ];
  const patient = <CardList cards={cards}/>;
  const tree = renderer.create(patient).toJSON();
  expect(tree).toMatchSnapshot();
});
