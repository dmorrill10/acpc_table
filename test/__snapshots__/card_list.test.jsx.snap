exports[`test with no cards works 1`] = `
<ol
  className="card-list" />
`;

exports[`test with two face-down cards works 1`] = `
<ol
  className="card-list">
  <li>
    <div
      className="card">
      <div
        className="logo" />
    </div>
  </li>
  <li>
    <div
      className="card">
      <div
        className="logo" />
    </div>
  </li>
</ol>
`;

exports[`test with two face-up cards works 1`] = `
<ol
  className="card-list">
  <li>
    <div
      className="card-face-up-h">
      <div
        className="top-left-rank">
        2
      </div>
      <div
        className="suit">
        &hearts;
      </div>
      <div
        className="bottom-right-rank">
        2
      </div>
    </div>
  </li>
  <li>
    <div
      className="card-face-up-d">
      <div
        className="top-left-rank">
        3
      </div>
      <div
        className="suit">
        &diams;
      </div>
      <div
        className="bottom-right-rank">
        3
      </div>
    </div>
  </li>
</ol>
`;
